library/acl.inc

```diff
@@ -828,7 +828,9 @@
             return false;
         elseif($count_user_allowed > 0)
             return true;
-        elseif($count_group_denied > 0)
+        // We do not exclude user that belongs to group A and B
+        // In case of A group have permissions and B group does not have permissions
+        elseif($count_group_denied > 0 && $count_group_allowed < 1)
```

Comment:
I have rechecked this issue and I still have problems when using a module permissions. in module permissions we use  zh_acl_check function .

the problem accrue when you give a USER two GROUPS in acl.

both GROUPS have all the acl permissions except admin.one GROUP doesn't have permission through the module management (access control) the person who belongs to both groups doesn't have permissions  as shown in the pictures below .


![4_admin](https://cloud.githubusercontent.com/assets/18376627/24491424/d4feb6fc-152f-11e7-80fc-b2158d0392a1.png)
![3_eyal](https://cloud.githubusercontent.com/assets/18376627/24491425/d5000c64-152f-11e7-9774-d77e01ef533c.png)
![2_eyal](https://cloud.githubusercontent.com/assets/18376627/24491426/d500bf10-152f-11e7-8062-00593f7f0824.png)
![1_eyal](https://cloud.githubusercontent.com/assets/18376627/24491423/d4fdd4e4-152f-11e7-9733-b00c5277e716.png)


Eyal will open a new branch and make a pull request fix to continue this discussion.